packages/CodingStandard/src/Fixer/Property/ArrayPropertyDefaultValueFixer.php

```diff
@@ -0,0 +1,103 @@
+<?php declare(strict_types=1);
+
+namespace Symplify\CodingStandard\Fixer\Property;
+
+use Nette\Utils\Strings;
+use PhpCsFixer\DocBlock\DocBlock;
+use PhpCsFixer\Fixer\DefinedFixerInterface;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use SplFileInfo;
+
+final class ArrayPropertyDefaultValueFixer implements DefinedFixerInterface
+{
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Array property should have default value, to prevent undefined array issues.',
+            []
+        );
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        // analyze only properties with comments
+        return $tokens->isAllTokenKindsFound([T_DOC_COMMENT, T_VARIABLE]);
+    }
+
+    public function isRisky(): bool
+    {
+        return false;
+    }
+
+    public function fix(SplFileInfo $file, Tokens $tokens): void
+    {
+        foreach ($tokens as $index => $token) {
```

Comment:
it is used in many fixers
it's not used when it's not needed (eg when fixer is not adding new content)

it is. your scenario works as you have it very small and few-elements offset is not big issue, as it means you not trying to fix end of script, which is usually class closing.

Imagine you would have 100 elements to fix in big crappy file. then fixer would fix only first ~80 properties